pkg/prometheus/resource_selector.go

```diff
@@ -584,3 +587,68 @@ func validateProberURL(url string) error {
 	}
 	return nil
 }
+
+// SelectScrapeConfigs selects ScrapeConfigs based on the selectors in the Prometheus CR and filters them
+// returning only those with a valid configuration.
+func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context) (map[string]*monitoringv1alpha1.ScrapeConfig, error) {
```

Comment:
```suggestion
func (rs *ResourceSelector) SelectScrapeConfigs() (map[string]*monitoringv1alpha1.ScrapeConfig, error) {
```

Looks like `ctx` isn't used here

Reply:
7a9402af3